### PR TITLE
[MIA-2077] Add Grape auth middleware

### PR DESCRIPTION
**Pull request by apokyn**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - name: Set up Bundler
+        run: gem install bundler:1.14.5
+      - name: Set up Dependencies
+        run: bundle install --path vendor/bundle
+      - name: Run specs
+        run: bundle exec rspec
+  release:
```
Comment (volodymyr-mykhailyk) [inline .github/workflows/main.yml]: Awesome that you added GitHub Actions. However, we should be pushing it to RubyGems. No worries though - I will update actions with next PR after this is merged and release a new version

File: README.md [hunk 2]
```diff
@@ -22,7 +20,30 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+####With Grape
```
Comment (volodymyr-mykhailyk) [inline README.md]: Awesome ❤️

File: lib/matic-jwt/grape/middleware/auth.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        def authenticate!
+          @env['auth_payload'] = authenticate_with_jwt!(secret)
+        end
+
+        def continue!
+          @app.call(@env)
+        end
+
+        def authenticate_with_jwt!(secret)
+          payload = jwt_authenticator.authenticate_with_secret!(secret)
```
Comment (volodymyr-mykhailyk) [inline lib/matic-jwt/grape/middleware/auth.rb]: Do I understand correctly that if request not authenticated - payload will be missing? I can come up with few usecases when we need payload regardless of authentication. However, they are not critical

File: .github/workflows/main.yml (hunk 1 above)
Comment (apokyn) [reply .github/workflows/main.yml]: Ok. thanks

File: lib/matic-jwt/grape/middleware/auth.rb (hunk 3 above)
Comment (apokyn) [reply lib/matic-jwt/grape/middleware/auth.rb]: Yes, you understood correctly. Will fix

Comment (apokyn) [reply lib/matic-jwt/grape/middleware/auth.rb]: Fixed

**Pull request merged**